File: BackEnd/api/model/user_model.py
from datetime import datetime
import peewee
from api.utils.operations_util import Utils


from api.utils.db import db

class User(peewee.Model):
    email = peewee.CharField(unique=True, index=True)
    username = peewee.CharField(unique=True, index=True)
    name = peewee.CharField()
    surname = peewee.CharField()
    registerDate = peewee.DateTimeField(default=datetime.now)
    updateDate = peewee.DateTimeField(default=datetime.now)
    userRole = peewee.CharField(default="user")
    password = peewee.CharField()

    class Meta:
        database = db

File: FrontEnd/src/app/models/user.model.ts
export class UserModel{
    
    idUser?: number;
    username: string;
    name: string;
    surname: string;
    registerDate: Date;
    updateDate: Date;
    userRole: string;
    constructor(idUser: number, username: string, name: string, surname: string, registerDate: Date, updateDate: Date, userRole: string){
        this.idUser = idUser;
        this.username = username;
        this.name = name;
        this.surname = surname;
        this.registerDate = registerDate;
        this.updateDate = updateDate;
        this.userRole = userRole;
    }
}